### PR TITLE
Adjustments for our wonderful Windows users

### PR DESCRIPTION
**Pull request by everett-toews**

### CODE REVIEW

**Comment by everett-toews** [comment]
@carolynvs @zack-shoylev Can you please give this a look?

FYI, I made these changes as this tutorial advises WIndows users to use the Docker Quickstart Terminal.

**Comment by carolynvs** [comment]
I am reviewing now, some of my comments are for the original, not just the changes. Trying to figure out how to leave comments on something other than the diff... :smile:

**Comment by carolynvs** [comment]
When I run `docker-compose up` I get the following error:

```
$ docker-compose up
ERROR: Validation failed in file 'C:\Users\caro8994\carina-examples\python-web-app\docker-compose-common.yml',
reason(s):
services.app.networks contains an invalid type, it should be an array
```

My docker-compose version is 1.6.0, build cdb920a. I'm updating now to see if that makes a difference.

**Comment by carolynvs** [comment]
Heh, updating to the latest version of toolbox broke everything... This may take a bit longer to fix and finish reviewing!

**Comment by everett-toews** [comment]
@carolynvs [This tweet](https://twitter.com/everett_toews/status/708047386380140546) fixed the Compose version issue :smile:

**Comment by carolynvs** [comment]
I've run into the dreaded https://github.com/docker/machine/issues/1954 which means that I can't test this for another 6 hours or so... I'm having another go at troubleshooting that problem while I'm waiting. :smile:

**Comment by carolynvs** [comment]
The instructions for https://github.com/everett-toews/getcarina.com/blob/py-dev-win/_getting-started/2016-03-08-python-web-app-dev.md#initialize-the-environment-1 are not what we recommend.

Can we point them instead to this? [Create and connect to a cluster](https://getcarina.com/docs/tutorials/create-connect-cluster/). In most of the tutorials that link is the first step, without any additional text.

**Comment by carolynvs** [comment]
For https://github.com/everett-toews/getcarina.com/blob/py-dev-win/_getting-started/2016-03-08-python-web-app-dev.md#build-the-images, I tested the interactive `docker login` command with both the QuickStart Terminal (aka git bash) and powershell, both worked just fine. So you can remove the "Note: On Windows, specify all of the information immediately for the docker login command. For example, docker login -u <docker-hub-username> -p <docker-hub-password> -e email@example.com" caveat.

**Comment by carolynvs** [comment]
I wasn't able to run through the tutorial due to unrelated problems but I've commented on anything that made my "windows sense" tingle.

**Comment by everett-toews** [comment]
>Can we point them instead to this? Create and connect to a cluster. In most of the tutorials that link is the first step, without any additional text.

I think this boils down to...do we need to add something like the following to [Create and connect to a cluster](https://getcarina.com/docs/tutorials/create-connect-cluster/)

If you've installed the Docker Toolbox on Windows and are using the Quickstart Terminal, run the following commands:

```bash
$ cd Downloads/mycluster
$ source docker.env
$ export DOCKER_CERT_PATH=C:\\Users\\everett\\Downloads\\mycluster
```

because you have to setup things a bit differently when using the Docker Toolbox on Windows and the Quickstart Terminal.

What do you think?

**Comment by carolynvs** [comment]
ick, okay yeah I remember this now...

How about we have them run this if they are on windows? We just need to get `/c/` changed to `c:/` and windows doesn't care about the remaining slashes, just the drive (which may not necessarily be `C:\`).

```
$ source docker.env
$ echo $DOCKER_CERT_PATH
/c/users/carolynvs/downloads/mycluster
$ export DOCKER_CERT_PATH=$(echo $DOCKER_CERT_PATH | sed 's/\///1' | sed 's/\//:\//1')
$ echo $DOCKER_CERT_PATH
c:/users/carolynvs/downloads/mycluster
$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
e51aea54d165        carina/consul       "/bin/consul agent -b"   28 hours ago        Up 28 hours                             af34ce2c-87a2-47c6-8022-4919ae9f552d-n1/carina-svcd
```

That works for me, how about it update Create and Connect with that and remove the caveat from this one?

**Comment by everett-toews** [comment]
I can make the update as part of this PR since this is what motivated it. I'll remove the caveat here.

**Comment by carolynvs** [comment]
:+1:

**Comment by everett-toews** [comment]
@carolynvs Give it one last look please. For the `sed` commands I changed the delimiters to `#` because I get confused looking at too many `\\/\/\\/\/\/\/\\///` in a row. o_O

**Comment by zack-shoylev** [comment]
I ran into some issues:

https://gist.github.com/zack-shoylev/0c5693956ad350377db5

After following the steps, at the step where I open the browser, I get a 502 Bad Gateway; the application log complains about not being able to connect upstream.

Also had to use DockerToolbox-1.10.3 ; the one I had installed I had to nuke and replace.

**Comment by everett-toews** [comment]
@zack-shoylev Ya. That's not a Windows issue. 

Did you alter the app at all? Why did you have to use `--force-recreate`?

**Comment by zack-shoylev** [comment]
No, I haven't altered the app. The first time it ran I stopped it and reset everything completely, so I thought it maybe didn't reset properly.

I can try to delete everything, reclone, and retry again

**Comment by everett-toews** [comment]
A `docker-compose down` should get you a clean slate.

**Comment by zack-shoylev** [comment]
Already tried that, but no luck.

**Comment by carolynvs** [comment]
@everett-toews Nice! I didn't know about changing the deliminator. Just tried it out and reviewed the updated files. Looks good!

**Comment by zack-shoylev** [comment]
I am guessing maybe [this](https://github.com/docker/compose/issues/2548) is the problem, but will have to check.

**Comment by zack-shoylev** [comment]
Yes, that was the problem. Mounting doesn't work from any random directory.

**Comment by everett-toews** [comment]
@zack-shoylev So what exactly do we need to recommend to Windows users to avoid that problem?

**Comment by zack-shoylev** [comment]
You need to clone carina-example in a directory under C:\Users\[Your username]

File: _getting-started/2016-03-08-python-web-app-dev.md [hunk 1]
```diff
@@ -157,21 +152,7 @@ When you are satisfied with your development and testing in your local developme
...
-
-    $ source docker.env
-    ```
-
-    If you used the `carina` CLI to create your cluster, source the environment variables with this commands.
-
-    ```bash
-    $ eval $(carina env mycluster)
-    ```
+1. Open a new terminal and initialize as per the instructions for the [Quickstart Termainal]({{ site.baseurl }}/docs/tutorials/create-connect-cluster/#quickstart-terminal)
```
Comment (zack-shoylev) [inline _getting-started/2016-03-08-python-web-app-dev.md]: small typo: Terminal

**Comment by carolynvs** [comment]
As I understand it, that is a quirk of VirtualBox, *not* Windows. So mac/linux users should be working in their home directory as well.

File: _getting-started/2016-03-08-python-web-app-dev.md (hunk 1 above)
Comment (everett-toews) [inline _getting-started/2016-03-08-python-web-app-dev.md]: "Termainal" is a perfectly cromulent word.

**Comment by everett-toews** [comment]
@carolynvs @zack-shoylev Thanks for the review!

**Pull request merged**